ambry-clustermap/src/main/java/com.github.ambry.clustermap/AmbryReplica.java

```diff
@@ -15,84 +15,58 @@
 
 import com.github.ambry.config.ClusterMapConfig;
 import com.github.ambry.utils.Utils;
-import java.io.File;
-import java.util.ArrayList;
 import java.util.List;
-import org.json.JSONObject;
+import java.util.Objects;
+import java.util.stream.Collectors;
 
-import static com.github.ambry.clustermap.ClusterMapSnapshotConstants.*;
 import static com.github.ambry.clustermap.ClusterMapUtils.*;
 
 
 /**
  * {@link ReplicaId} implementation to use within dynamic cluster managers.
  */
-class AmbryReplica implements ReplicaId {
+abstract class AmbryReplica implements ReplicaId {
   private final AmbryPartition partition;
-  private final AmbryDisk disk;
   private final long capacityBytes;
   private volatile boolean isSealed;
-  private volatile boolean isStopped;
-  private final ResourceStatePolicy resourceStatePolicy;
+  volatile boolean isStopped;
+  final ResourceStatePolicy resourceStatePolicy;
```

Comment:
I wonder if `onReplicaUnavailable()` and `onReplicaResponse()` in this class are really needed for `CloudServiceReplica`. If not, we can move it into `AmbryServerReplica`.

Reply:
I suppose they are at least needed as abstract methods in AmbryReplica since `onReplicaEvent` calls them and it would be unclean IMO to have `onReplicaEvent` have to check the replica type and cast before calling the method.

Whether the `ResourceStatePolicy` is needed for cloud replicas is another question. One could just make the CloudServiceReplica always up and not have any failure tracking. For our preliminary testing environment I am inclined to leave it in and let the CloudServiceReplica go "down" when faced with service unavailability and see how it behaves. We definitely would need to take a second look at this though.